src/Dialogs/Composer/Dialog.vala

```diff
@@ -190,14 +197,16 @@ public class Tuba.Dialogs.Compose : Adw.Window {
 	}
 
 	[GtkCallback] void on_commit () {
-		//working = true
+		if (working)
+			return; // avoid double send
```

Comment:
I don't think on_commit can be called as long as the composer is insensitive but it doesn't hurt I guess!

I'll leave it up to you to decide whether to keep it or not, but if you don't then I'd suggest getting rid of `working` property altogether as it can be replaced with the snippet I suggested prior

Reply:
Huh, yeah, turns out that on_commit is completely safe once the composer is insensitive, even when using the Ctrl+Return accelerator!